Remove tftpsync package

m-czernek opened the pull request:
## What does this PR change?

Since SUMA 5.2 and newer doesn't support 4.3 proxies anymore, this PR removes legacy code used for TFTP sync to 4.3.x proxies. 

Note, I don't now if we need some special changelog. I will ping relengs when this PR gets a green +1.

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/27980

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

rjmateus commented:
By removing this package we should also consider removing some salt formulas that use it to synchronize images to proxies.
@aaannz for awareness

aaannz commented:
AFAIK there are no salt formulas, but there is a setup code in the proxy containers no longer needed.

For context, that setup code in the container is there only to prevent user seeing errors on `cobbler sync`, but synced data were not actually used.

See https://github.com/uyuni-project/uyuni/blob/master/containers/proxy-httpd-image/uyuni-configure.py#L157 and https://github.com/uyuni-project/uyuni/blob/master/containers/proxy-httpd-image/uyuni-configure.py#L176

m-czernek commented:
I appreciate y'all taking a look! I have removed outdated configuration, and added context of what to do after +1 from engineering into the PR description.

ycedres reviewed (dismissed):
LGTM.

ktsamis reviewed (dismissed):
I think it's fine to remove the tests in master yes. But let's leave the gh PR tests to run to see the results

Pablogoliva commented on testsuite/features/step_definitions/command_steps.rb:
```diff
@@ -677,23 +677,6 @@
   node.run('rm /tmp/test_user_defined_state')
 end
 
-# TODO: remove this step definition when we deprecate the non-containerized components
```
You are deleting the steps but not the features by itself. It is going to break our testsuite.

Pablogoliva reviewed (changes requested):
You have to remove the features where the steps you are deleting are used, otherwise the testsuite is not going to work in the next execution. And that's exactly what's happening in the GH actions tests. Let me point you to what has to be deleted.

Pablogoliva commented:
> You have to remove the features where the steps you are deleting are used, otherwise the testsuite is not going to work in the next execution. And that's exactly what's happening in the GH actions tests. Let me point you to what has to be deleted.

Clarifed with Marek, it's not breaking anything.
